File: frontend/src/api/sources.ts
// ponytail: native fetch for sources CRUD
const base = '/api/sources';

export const getSources = async (sessionId: string) => 
  (await fetch(`${base}/${sessionId}`)).json();

export const saveSources = async (sessionId: string, sources: any[]) => 
  (await fetch(base, {
    method: 'POST',
    headers: { 'Content-Type': 'application/json' },
    body: JSON.stringify({ session_id: sessionId, sources })
  })).json();

export const clearSources = async (sessionId: string) => 
  (await fetch(`${base}/${sessionId}`, { method: 'DELETE' })).json();

File: backend/app/api/routes/sources.py
from fastapi import APIRouter
from pydantic import BaseModel
from typing import List, Dict, Any

router = APIRouter()

# ponytail: simple in-memory store for sources
_sources_db: Dict[str, List[Dict[str, Any]]] = {}

class SourcesRequest(BaseModel):
    session_id: str
    sources: List[Dict[str, Any]]

@router.get("/sources/{session_id}")
async def get_sources(session_id: str):
    return {"status": "success", "data": _sources_db.get(session_id, [])}

@router.post("/sources")
async def save_sources(req: SourcesRequest):
    _sources_db[req.session_id] = req.sources
    return {"status": "success"}

@router.delete("/sources/{session_id}")
async def clear_sources(session_id: str):
    _sources_db.pop(session_id, None)
    return {"status": "success"}
